Source/Csla/Core/ContextDictionary.cs

```diff
@@ -76,25 +76,34 @@ void IMobileObject.SetChildren(SerializationInfo info, MobileFormatter formatter
     /// <inheritdoc cref="System.Collections.IDictionary.IsReadOnly"/>
     public bool IsReadOnly
     {
-      get => ((IDictionary)this).IsReadOnly;
+      get => false;
     }
 
     /// <inheritdoc cref="System.Collections.IDictionary.IsFixedSize"/>
     public bool IsFixedSize
     {
-      get => ((IDictionary)this).IsFixedSize;
+      get => false;
     }
 
     /// <inheritdoc cref="System.Collections.IDictionary.Add(object, object?)"/>
     public void Add(object key, object value)
     {
-      ((IDictionary)this).Add(key, value);
+      bool added = TryAdd(key, value);
+      if (!added)
+      {
+        throw new ArgumentException("An item with the same key has already been added.");
+      }
     }
 
     /// <inheritdoc cref="System.Collections.IDictionary.Remove(object)"/>
     public void Remove(object key)
     {
-      ((IDictionary)this).Remove(key);
+      object dummy;
+      var removed = TryRemove(key, out dummy);
```

Comment:
`out var _` is a bit more effcient because the compiler can discard the variable completly.

Reply:
Changed

Reply:
@Bowman74 don't mark conversations as resolved - the reviewer should do that